odilia/src/state.rs

```diff
@@ -84,6 +85,33 @@ impl ScreenReaderState {
 		})
 	}
 
+	pub async fn get_or_create_atspi_cache_item_to_cache(&self, atspi_cache_item: atspi::cache::CacheItem) -> OdiliaResult<CacheItem> {
+		let prim = atspi_cache_item.object.clone().try_into()?;
+		if self.cache.get(&prim).await.is_none() {
+			self.cache.add(
+				CacheItem::from_atspi_cache_item(
+					atspi_cache_item,
+					Arc::clone(&self.cache),
+					self.atspi.connection(),
+				).await?
+			).await;
+		}
+		self.cache.get(&prim).await.ok_or(CacheError::NoItem.into())
+	}
+	pub async fn get_or_create_event_object_to_cache<T: Signified>(&self, event: &T) -> OdiliaResult<CacheItem> {
+		let prim = AccessiblePrimitive::from_event(event)?;
+		if self.cache.get(&prim).await.is_none() {
+			self.cache.add(
+				CacheItem::from_atspi_event(
+					event,
+					Arc::clone(&self.cache),
+					self.atspi.connection(),
+				).await?
+			).await;
+		}
+		self.cache.get(&prim).await.ok_or(CacheError::NoItem.into())
```

Comment:
FYI dashmap also mimics the `entry` API like the std hashmap, which can be nice. But this looks good too!